[BUG] panic: io.Copy will get stuck (DiskPiece.ReadAt swallows error)

# [BUG] panic: io.Copy will get stuck (DiskPiece.ReadAt swallows error)

## Summary

`DiskPiece.ReadAt` in `server/torr/storage/torrstor/diskpiece.go` swallows the error returned by
`ff.ReadAt`. When the piece file exists but is empty/partial at the requested offset, `os.File.ReadAt`
returns `(0, io.EOF)`, but `DiskPiece.ReadAt` forces `return n, nil` — discarding the error. The
anacrolix/torrent storage wrapper (`storage/wrappers.go`) then sees `(n=0, err=nil)` and triggers its
defensive panic to prevent an infinite `io.Copy` loop, **crashing the whole TorrServer process**.

This is the same class of issue as [#430](https://github.com/anacrolix/torrent/issues/430) (2020),
where the fix was to correctly return `io.EOF`. `MemPiece.ReadAt` already does the right thing
(`if n == 0 { return 0, io.EOF }`); only `DiskPiece` is affected.

## Reproduce

- Config: `UseDisk: true`, `TorrentsSavePath` on an external NTFS volume (Keenetic tntfs), aarch64
  static build (MatriX.142).
- Trigger: active streaming of a torrent with enough peer activity to trigger piece eviction
  (`cleanPieces`) during hashing. On my setup it crashed **~5 times per hour** during playback.

## Crash trace (GOTRACEBACK=crash)

```
panic: io.Copy will get stuck

goroutine 159748 [running]:
panic(...)
github.com/anacrolix/torrent/storage.Piece.ReadAt        storage/wrappers.go:88
github.com/anacrolix/torrent/storage.(*Piece).ReadAt     <autogenerated>
io.(*SectionReader).Read                                 io/io.go:516
io.copyBuffer → io.Copy                                  io/io.go:388
github.com/anacrolix/torrent.(*Torrent).hashPiece        torrent.go:747
```

(38 goroutines in the full dump — MSE cipher readers, UTP conns, `torrstor.DiskPiece.ReadAt`/
`cleanPieces`/`cache.clearPriority`, rate.Limiter waits — typical for an active client; no other
panic frame.)

## Root cause

```go
// server/torr/storage/torrstor/diskpiece.go
func (p *DiskPiece) ReadAt(b []byte, off int64) (n int, err error) {
    ...
    n, err = ff.ReadAt(b, off)        // may return (0, io.EOF) for empty/partial piece file
    ...
    return n, nil                     // BUG: err discarded -> anacrolix panics on (0, nil)
}
```

The race: a `DiskPiece` file exists but has no data at `off` (e.g. eviction/cleanup window while
anacrolix believes the piece is complete and tries to hash it). `ff.ReadAt` correctly reports
`io.EOF`; the wrapper turns it into `(0, nil)` and the anacrolix guard panics.

## Fix

One line — propagate the error so the anacrolix wrapper sees `io.EOF`, calls `MarkNotComplete()`,
and re-downloads the piece instead of crashing:

```diff
--- a/server/torr/storage/torrstor/diskpiece.go
+++ b/server/torr/storage/torrstor/diskpiece.go
@@ -71,7 +71,7 @@ func (p *DiskPiece) ReadAt(b []byte, off int64) (n int, err error) {
 	if int64(len(b))+off >= p.piece.Size {
 		go p.piece.cache.cleanPieces()
 	}
-	return n, nil
+	return n, err
```

`MemPiece.ReadAt` already returns `io.EOF` on the zero-read path, so only `DiskPiece` needs the change.

## Validation

Built `MatriX.142` + this one-line fix (linux/arm64) and deployed. The exact torrent that crashed
5×/hour streamed for 40 s (and counting) with a **stable PID, flat crash log, zero panics**. Before
the fix, the same torrent crashed within minutes. 24 h monitoring in progress.

## Environment

- TorrServer: MatriX.142 (built from master @ `0708143`)
- Platform: linux/arm64 (Keenetic Netcraze, aarch64)
- Storage: external NTFS via tntfs, `UseDisk: true`
- anacrolix/torrent replace: `github.com/tsynik/torrent v1.2.25`

Happy to open a PR with the patch if helpful.

## Pull request

Fix submitted in the linked PR (ix/diskpiece-eof). Minimal one-line change to server/torr/storage/torrstor/diskpiece.go.

--- a/server/torr/storage/torrstor/diskpiece.go
+++ b/server/torr/storage/torrstor/diskpiece.go
@@ -71,7 +71,7 @@ func (p *DiskPiece) ReadAt(b []byte, off int64) (n int, err error) {
 	if int64(len(b))+off >= p.piece.Size {
 		go p.piece.cache.cleanPieces()
 	}
-	return n, nil
+	return n, err
 }
 
 func (p *DiskPiece) Release() {

